lucille-core/src/main/java/com/kmwllc/lucille/connector/storageclient/TraversalParams.java

```diff
@@ -0,0 +1,95 @@
+package com.kmwllc.lucille.connector.storageclient;
+
+import static com.kmwllc.lucille.connector.FileConnector.GET_FILE_CONTENT;
+import static com.kmwllc.lucille.connector.FileConnector.HANDLE_ARCHIVED_FILES;
+import static com.kmwllc.lucille.connector.FileConnector.HANDLE_COMPRESSED_FILES;
+import static com.kmwllc.lucille.connector.FileConnector.MOVE_TO_AFTER_PROCESSING;
+import static com.kmwllc.lucille.connector.FileConnector.MOVE_TO_ERROR_FOLDER;
+
+import com.kmwllc.lucille.core.fileHandler.FileHandler;
+import com.typesafe.config.Config;
+import java.net.URI;
+import java.util.List;
+import java.util.Objects;
+import java.util.regex.Pattern;
+
+/**
+ * The arguments / parameters associated with a traversal on a StorageClient.
+ */
+public class TraversalParams {
+
+  // provided arguments
+  public final URI pathToStorageURI;
+  public final String docIdPrefix;
+  private final List<Pattern> excludes;
+  private final List<Pattern> includes;
+  public final Config fileOptions;
+
+  // derived arguments
+  public final boolean getFileContent;
```

Comment:
pls. create getters for all instance variables we need to access and make the variables themselves private